frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -978,6 +1008,38 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Used to delegate a new destination to earn the stake rewards.
+        /// Possible situations:
+        /// 1 - Staker is the delegate and want to delegate someone else;
+        /// 2 - Delegate is not the staker and want to delegate someone else;
+        /// 3 - Staker is not the delegate but want to be.
+        /// Main dificulty here is because is not possible at first to know if "origin" is the staker, the delegate or someone else.
+        #[pallet::weight(T::DbWeight::get().reads_writes(1, 1))]
+        pub fn set_new_delegate(
+            origin: OriginFor<T>,
+            staker: T::AccountId,
+            contract_id: T::SmartContract,
+            destination_account: T::AccountId
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            let delegate = Delegates::<T>::get(&staker, &contract_id).ok_or(Error::<T>::StakingDelegateDontExist)?;
```

Comment:
Why is this an error? Shouldn't this be the default state?

Reply:
The purpose of this error message is to indicate that the specified 'staker' is not staking on the given 'contract_id', as there is no matching entry in the 'Delegates' storage for the 'staker' -> 'contract_id' -> 'delegate' relationship. This relationship is normally established automatically by the 'bond_and_stake' function when a staker bonds and stakes tokens on a contract. Therefore, this error message indicates that the staker is not currently staking on the contract, and may need to take action to correct the issue.